src/app/core/services/opa/service.ts

```diff
@@ -144,11 +143,13 @@ export class OPAService {
     return this._http.delete(url);
   }
 
-  get violationPageIndex(): number {
-    return this._violationPageIndex;
+  getViolationPageIndex(projectId: string, clusterId: string, violationIndex: number): number {
+    const id = `${projectId}-${clusterId}-${violationIndex}`;
+    return this._violationPageIndex.get(id);
   }
 
-  saveViolationPageIndex(index: number): void {
-    this._violationPageIndex = index;
+  saveViolationPageIndex(projectId: string, clusterId: string, violationIndex: number, pageIndex: number): void {
+    const id = `${projectId}-${clusterId}-${violationIndex}`;
```

Comment:
Are we sure that the `violationIndex` will not change if the number of violations will change? I think it would be better to use constraint name (is it unique?) and the violation name instead and keep the project and cluster IDs.

Reply:
It isn't connected to the violation count, but you're right, I think using the constraint name is better. Checked and I can't create a constraint with the same name in this cluster. We cannot really use a violation name, because which one do you would choose out of the list (we need the paginator for this list)? I think project-cluster-constraint is as explicit as we could be. Will change it :)

Reply:
@maciaszczykm done

Reply:
Yeah, not violations but constraint count. Violation name is also out of scope.